src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,262 @@
+            } else if (userCommand.startsWith("deadline")) {
+                addDeadline(userCommand);
+            } else if (userCommand.startsWith("delete")) {
+                deleteTask(userCommand);
+            } else {
+                printSign();
+                System.out.println(" ☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+                printSign();
+            }
+        } while (!Objects.equals(userCommand, "bye"));
+        exit();
+        saveData();
+    }
+
+    public static void loadData(Path dataPath) throws IOException {
+        File dataDirectory = new File(dirPath.toString());
+
+        if(!dataDirectory.exists()) {
+            dataDirectory.mkdir();
+        }
+
+        File dataFile = new File(dataPath.toString());
+        if (!dataFile.exists()) {
+            dataFile.createNewFile();
+        }
+
+        Scanner dataScanner = new Scanner(dataFile);
+        while (dataScanner.hasNext()) {
+            String content = dataScanner.nextLine();
+            String taskType = content.substring(0,1);
+            String taskContent = content.substring(8);
+            int taskStatusIndex = 4;
+            int taskDateIndex = taskContent.indexOf('|');
+            if (taskType.equals("T")) {
+                tasks.add(new Todo(taskContent));
+                taskNum++;
+                if (content.charAt(taskStatusIndex) == '1') {
+                    tasks.get(tasks.size() - 1).markAsDone();
+                }
+            } else if (taskType.equals("D")) {
+                tasks.add(new Deadline(taskContent.substring(0, taskDateIndex - 1), taskContent.substring(taskDateIndex + 2)));
+                taskNum++;
+                if (content.charAt(taskStatusIndex) == '1') {
+                    tasks.get(tasks.size() - 1).markAsDone();
+                }
+            } else if (taskType.equals("E")) {
+                tasks.add(new Event(taskContent.substring(0, taskDateIndex - 1), taskContent.substring(taskDateIndex + 2)));
+                taskNum++;
+                if (content.charAt(taskStatusIndex) == '1') {
+                    tasks.get(tasks.size() - 1).markAsDone();
+                }
+            }
+        }
+    }
+
+    public static void saveData() {
+        try {
+            FileWriter writer = new FileWriter(dataPath.toString());
+            for (Task task : tasks) {
+                if (task instanceof Todo) {
+                    writer.write("T | " + task.getStatusIcon() + " | " + task.getDescription() + System.lineSeparator());
+                } else if (task instanceof Deadline) {
+                    writer.write("D | " + task.getStatusIcon() + " | " + task.getDescription() + " | " + ((Deadline) task).getBy() + System.lineSeparator());
+                } else if (task instanceof Event) {
+                    writer.write("E | " + task.getStatusIcon() + " | " + task.getDescription() + " | " + ((Event) task).getAt() + System.lineSeparator());
```

Comment:
Some of these lines exceed the convention of [120 characters per line](https://se-education.org/guides/conventions/java/basic.html#layout). How about reformatting the String that will be written differently? E.g. Assigning variables to parts of the String, using [Java String format()](https://www.javatpoint.com/java-string-format) method.